### ISSUE TITLE
scripts/make/configure.mk introduces insecure RPATH

### ISSUE THREAD

**Issue by dvzrv** [sadko4u/lsp-plugins#60]
When building lsp-plugins as a package for Arch Linux I have to manually remove [the hardcoded RPATH](https://github.com/sadko4u/lsp-plugins/blob/master/scripts/make/configure.mk#L85) set in `scripts/make/configure.mk`.
Hardcoding it is problematic, as it allows for loading libs from non-default locations.

Maybe a better way can be found to set `LD_PATH` based on user request?

**Comment by sadko4u** [comment]
This option originally was added because I had problems cross-building i586 binaries from x86_64 system on an openSUSE distribution. It just couldn't find some 32-bit libraries even if they've been installed.
So probably yes, it's not an elegant (and more probably an ugly solution). So if you have ideas about how we can avoid requirement to set rpath, I'll introduce it.

**Comment by sadko4u** [comment]
Created **github-issue-60** branch.
Seems that this commit will fix the hard-coded RPATH value:
https://github.com/sadko4u/lsp-plugins/commit/77c93ad1720693094c7c2a3fdb1a5ad0589220df

On my systems, the build is OK for all build profiles, without extra linker errors.
Please verify

**Comment by dvzrv** [comment]
@sadko4u thanks for getting back to me so quickly!

The commit in question fixes the RPATH issue, but reintroduces [missing full RELRO](#4 ) for the ladspa, the (main) lv2 and the vst plugins.

I usually reverted to using sed on the `LD_PATH` line in question and e.g. removing `:/usr/local/lib`

**Comment by sadko4u** [comment]
> The commit in question fixes the RPATH issue, but reintroduces missing full RELRO for the ladspa, the (main) lv2 and the vst plugins.
Hmmm, that's interesting and normally should not happen. I'll inspect the changes.

**Comment by sadko4u** [comment]
@dvzrv 
I've just inspected my local build with **hardening-check** tool:
It shows:
```
Position Independent Executable: yes
Read-only relocations: yes
```
for all executable binaries

And 
```
Position Independent Executable: no, regular shared library (ignored)
Read-only relocations: yes
```
for all shared objects.

Maybe there's something wrong with your build?

**Comment by sadko4u** [comment]
@dvzrv 
Any suggestions?

**Comment by dvzrv** [comment]
> Position Independent Executable: no, regular shared library (ignored)

Doesn't that mean, the application doesn't check this for shared libraries?

Well, the tool we use for checking during packaging is [namcap](https://wiki.archlinux.org/index.php/Namcap).
The output is attached here:
[lsp-plugins-1.1.11-check.log](https://github.com/sadko4u/lsp-plugins/files/3994802/lsp-plugins-1.1.11-check.log)

It also indicates [overlinking](https://wiki.mageia.org/en/Overlinking_issues_in_packaging) of the standalones with `libpthread.so` and some of the lv2 and all vst plugins with `libpthread.so` and `libgcc_s.so` (that is also the case without the patch though!).

**Comment by dvzrv** [comment]
I think this might be the culprit: https://github.com/sadko4u/lsp-plugins/commit/77c93ad1720693094c7c2a3fdb1a5ad0589220df#r36549134

**Comment by sadko4u** [comment]
Seems that you're right.
Pushed the change. Please try again.

**Comment by sadko4u** [comment]
> It also indicates overlinking of the standalones with libpthread.so
Pushed additional changes. Seems to be fixed now.

>and libgcc_s.so
Hmm. I don't add any extra flags for this library.

**Comment by dvzrv** [comment]
Hm, this doesn't apply cleanly ontop of 1.1.11 anymore, but I trust that you did the right thing! :)

**Comment by dvzrv** [comment]
That being said, further overlinking can be solved in the future. I think fixing the issues with 1.1.11 is already a big thing :)

**Comment by sadko4u** [comment]
Nice, I'll merge the changes into **devel** branch then.

**Comment by sadko4u** [comment]
Merged changes into **devel** branch: 63c2c5a59112f0d941ca70aca8d312e9e056a86e

**Comment by sadko4u** [comment]
Removed **github-issue-60** branch.

**Comment by sadko4u** [comment]
Available in 1.1.13 release!
